### DISCUSSION TITLE
Can c3c build c++ files?

### DISCUSSION THREAD

**Discussion by felixf4xu** [c3lang/c3c#1576] [General]
Hi,

I see c3c can build c source files, 
```jsonc
  // C sources if the project also compiles C sources
  // relative to the project file.
  // "c-sources": [ "csource/**" ],
  // Include directories for C sources relative to the project file.
  // "c-include-dirs": [ "csource/include" ],
```
I'm just curious, can c3c build c++ files?

I created a c++ file and use `extern "C"` to write c interface functions, it seems to build ok but it can't be linked.

**Comment by lerno** [comment]
c3c can't compile C or C++, but it can invoke a C compiler and then link automatically the resulting object files. Currently C++ files are not picked up I think(?) but could be.
